### DISCUSSION TITLE
Search by ASN

### DISCUSSION THREAD

**Discussion by darmiel** [jonaswinkler/paperless-ng#664] [Archived FRs]
Hello!
It has happened to me several times now that I wanted to view a document in Paperless (e.g. for emailing), for which I knew the ASN. It would be nice to be able to search directly for an ASN in the document overview to get to the digitized documents faster. Maybe something like that:
![Screenshot 2021-02-28 at 20 05 10](https://user-images.githubusercontent.com/71837281/109430179-461fa600-7a00-11eb-8000-2d2c31b0259f.png)

**Comment by shamoon** [comment]
Doesn’t the top search field work with ASN? I don’t use it. Seems like if you’re looking for one exact doc the top bar is a better choice (conceptually you’re searching for one doc, not filtering for a bunch)

**Reply by darmiel** [reply]
No, neither in the "Filter" text field, nor in the "Search Documents" text field I can search for ASNs. I have tried to enter:
* {ASN}
* #{ASN}
* ASN {ASN}

without success.

**Reply by jonaswinkler** [reply]
Hm. That's a clear oversight on my part. I'll think of something. Currently, there's no way to search for a specific ASN.

I'll certainly add that to the full text search. I'm also trying to wrap my head around integrating the full text search with the document list better (so that filtering and bulk modifications are possible on search results).

I'm very hesitant to add a text field for ASN searching, since this entire ASN feature isn't used by a lot of users (there was a survey about a month ago, and about 20% users out of a total of 100 responses use that). Some even requested to hide ASN-specific elements.

**Reply by darmiel** [reply]
Yes, I understand that.

Since one of the last updates, there is a dropdown at the filter where you can select if you want to search in the content as well. What if you could filter by ASN there?
(something like:)
![Screenshot 2021-03-08 at 16 34 08](https://user-images.githubusercontent.com/71837281/110342933-3faab300-802c-11eb-8161-3c537211f4a0.png)

If that would be okay with you, I could create a PR with the ASN filter. Because I have already changed it locally. 😁

**Reply by jonaswinkler** [reply]
@darmiel Yes. Good idea! Make a PR.

**Reply by darmiel** [reply]
Done 😁 https://github.com/jonaswinkler/paperless-ng/pull/726
